cartographer_ros/cartographer_ros/node.cc

```diff
@@ -466,51 +466,53 @@ bool Node::ValidateTopicNames(
   return true;
 }
 
+cartographer_ros_msgs::StatusResponse Node::CheckTrajectoryState(
+    int trajectory_id, std::set<TrajectoryState> valid_states) {
+  const auto trajectory_states = map_builder_bridge_.GetTrajectoryStates();
+  cartographer_ros_msgs::StatusResponse status_response;
+
+  if (!(trajectory_states.count(trajectory_id))) {
+    status_response.message =
+        absl::StrCat("Trajectory ", trajectory_id, " doesn't exist.");
+    status_response.code = cartographer_ros_msgs::StatusCode::NOT_FOUND;
+    return status_response;
+  }
+
+  const auto trajectory_state = trajectory_states.at(trajectory_id);
+  const std::map<TrajectoryState, std::string> state_to_string = {
```

Comment:
have helper function TrajectoryStateToString that uses switch without default so the compiler enforces the mapping to be complete